### ISSUE TITLE
[4.0] Articles disappearing if length of text is too long

### ISSUE THREAD

**Issue by nozaro33** [joomla/joomla-cms#34377]
### Steps to reproduce the issue

Can use the text here as reference: https://whiletrue.neocities.org/lte.html
Copy and paste twice into Text editor, and save.
Should get error message, but article is still saved to db, yet no longer viewable in Article Manager

### Expected result

Article should either not be saved with error, or should allow longer text

### Actual result

You get an error if it's in non-latin characters.
<img width="984" alt="Screen Shot 2021-06-03 at 20 46 57" src="https://user-images.githubusercontent.com/25219289/120649777-b6cad880-c4af-11eb-966b-afdc2dd30dbb.png">
or you may get this error if it's in English
<img width="1250" alt="Screen Shot 2021-06-03 at 20 53 53" src="https://user-images.githubusercontent.com/25219289/120649833-c4805e00-c4af-11eb-8a9f-9c374e9711a8.png">
(Finder token error will also show when I did the text in Joomla 3.9.27, but you can still see the article in article manager)

Articles will no longer show in the backend, but will still show normally in frontend. This issue persists regardless of language used.

### System information (as much as possible)

Joomla 4 RC 1
PHP 7.4.2
Mysql 5.7.26

### Additional comments
As you can see, ID 4,5,7 are no longer visible in backend. Please note that they still show up fine in frontend.
Note:  Even if you remove the "long text" from phpmyadmin, the article is still not visible from the backend Articles manager

<img width="964" alt="Screen Shot 2021-06-03 at 20 54 27" src="https://user-images.githubusercontent.com/25219289/120650263-2fca3000-c4b0-11eb-80cd-ffed6268ad77.png">
<img width="735" alt="Screen Shot 2021-06-03 at 20 55 49" src="https://user-images.githubusercontent.com/25219289/120650277-32c52080-c4b0-11eb-9324-f388ad184459.png">

**Comment by nikosdion** [comment]
I cannot reproduce this with a. the text you specified; b. 150 paragraphs of Lorem Ipsum which actually end up with more words and total size than your example and c. randomly adding Emojis.

Reading your error messages, it doesn't look like the article text itself is the problem. I've created articles far longer than 75KB (your test text).

The first error message talks about the meta description having invalid characters. The message makes me think that your table collation is wrong. If you're using a non-UTF8 collation on your tables they will fail to store most non-ASCII characters, depending on which collation is actually used. Check the collation on the tables. Joomla tries to install with UTF8MB4 if possible, or UTF8 is it's not supported. If the database doesn't support either all bets are off...

The second error message says that the in-memory Finder tokens table is full, i.e. MySQL has ran out of memory. When Finder fails to run the UCM entry for the article is not created which may be why you don't see the article listed. You'd need to let MySQL use more memory in the `my.cnf` settings.

Basically, these look like issues with the database server, not Joomla per se.

**Comment by nozaro33** [comment]
@nikosdion  Yes, my databases are UTF8MB4, and using the exact same MySQL environment, I don't get the error with Joomla 3.9.27 using non-ASCII  characters. I'll try to reproduce it in another server, maybe something wrong with my local setup

**Comment by nikosdion** [comment]
Also check the encoding your browser is using when displaying the page. The only time I had the first problem in the past (in my own extensions, not Joomla) is when the table field was the wrong collation (utf8_general_ci instead of one of the supported utf8mb4 collations).

The latter problem is definitely something you need to address in MySQL configuration. Frankly, I don't know why Joomla is still using MEMORY tables for Finder when [the MySQL documentation](https://dev.mysql.com/doc/refman/5.7/en/memory-storage-engine.html) says they don't perform any better than InnoDB. More so when using a MEMORY table (as opposed to TEMPORARY) means that once the memory space is up the table fails instead of being converted to an on-disk format. It sounds like a bad idea for indexing large sites.

**Comment by PhilETaylor** [comment]
Im going to put my money on this being a com_finder issue... you know that the finder effectively stores a whole copy of the content too... paging @Hackwar

**Comment by PhilETaylor** [comment]
Take the text from https://whiletrue.neocities.org/lte.html and paste it in 20 times the the body of the article... then save... 

I get 

<img width="842" alt="Screenshot 2021-06-03 at 17 07 24" src="https://user-images.githubusercontent.com/400092/120676669-2fb93880-c48e-11eb-86eb-b065b894c770.png">

**Comment by nikosdion** [comment]
Yeah, that's because Finder's token table is MEMORY which is limited by the MySQL server configuration. This table should really have been InnoDB. I have a really big limit on my setup because of the weird things I have to try which explains why I'm not hitting it even with an ~1MB article.

**Comment by nikosdion** [comment]
Also worth noting, MEMORY tables have fixed record length which makes them an even worse candidate for a table like Finder's tokens. We're basically burning through memory which mostly sits unused...

**Comment by PhilETaylor** [comment]
changing the table to InnoDB and then I get :) 

<img width="768" alt="Screenshot 2021-06-03 at 17 11 34" src="https://user-images.githubusercontent.com/400092/120677256-cb4aa900-c48e-11eb-96a1-9c2e87a52d76.png">

**Comment by nozaro33** [comment]
@PhilETaylor  I've gotten both the errors before, but do you encounter the issue that you can't see the "long text" article in the backend of article manager? (it just vanishes) Or is that issue just some random problem on my end.

**Comment by ReLater** [comment]
> Or is that issue just some random problem on my end.

It's far away of beeing a "random problem". I've seen this error message often enough and it was reported several times in forums. It simply depends on the power of your server and database settings and so on when and if your Joomla collapses and how many of your work will be lost if you write a very long article. And it depends also on how many content you already have.

And I'm sure that it was already reported that Joomla 4 then doesn't save the article while Joomla 3 saves it.

**Comment by nozaro33** [comment]
@ReLater  Well, the problem I'm encountering is Joomla 4 still saves the article, you just can't see it anymore in the backend. (But shows up fine in the front!) I don't encounter this (article disappearing) issue in Joomla 3.9.27

**Comment by ChristineWk** [comment]
Relevant? #35701 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34377">issues.joomla.org/tracker/joomla-cms/34377</a>.</sub>

**Comment by Quy** [comment]
Please test PR #35995

**Comment by richard67** [comment]
> Please test PR #35995

@Quy Does that PR solve the issue so we can close it?

**Comment by Hackwar** [comment]
#36749 should fix this.

**Comment by richard67** [comment]
Closing as having a pull request. Please test #36749 . Thanks in advance.
